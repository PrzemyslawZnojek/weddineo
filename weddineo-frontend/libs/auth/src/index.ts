export * from './lib/auth.module';
export * from './lib/+state/auth.facade'
