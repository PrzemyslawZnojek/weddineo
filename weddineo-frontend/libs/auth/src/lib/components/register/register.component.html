<div [fxFlex]="'100%'" [fxLayoutAlign]="'center center'" class="container">
  <mat-card [fxFlex.gt-xs]="'500px'" [fxFlex]="'80%'">

    <mat-card-content [fxLayout]="'column'">
      <weddineo-frontend-text-input
        [label]="'auth.register.username' | translate"
        [fc]="formGroup.get('username')"
        [errors]="errors"
      >
      </weddineo-frontend-text-input>
      <weddineo-frontend-text-input
        [label]="'auth.register.email' | translate"
        [fc]="formGroup.get('email')"
        [errors]="errors"
      >
      </weddineo-frontend-text-input>
      <weddineo-frontend-text-input
        [label]="'auth.register.password' | translate"
        [fc]="formGroup.get('password')"
        [type]="'password'"
        [errors]="errors"
      >
      </weddineo-frontend-text-input>
      <weddineo-frontend-text-input
        [label]="'auth.register.passwordConfirm' | translate"
        [fc]="formGroup.get('passwordConfirm')"
        [type]="'password'"
        [errors]="errors"
      >
      </weddineo-frontend-text-input>
      <div [fxLayout]="'row'" [fxLayoutGap]="'15px'">
        <weddineo-frontend-action-button
          [label]="'auth.register.register' | translate"
          [width]="'150px'"
          [fxLayoutAlign]="'center center'"
          (onClick)="register()"
        >
        </weddineo-frontend-action-button>
        <weddineo-frontend-action-button
          [color]="'secondary'"
          [label]="'auth.login.signIn' | translate"
          [width]="'150px'"
          [fxLayoutAlign]="'center center'"
          (onClick)="goToLogin()"
        >
        </weddineo-frontend-action-button>
      </div>
    </mat-card-content>
  </mat-card>
</div>
