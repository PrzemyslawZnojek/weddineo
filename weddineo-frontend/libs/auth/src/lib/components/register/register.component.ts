import { Component, OnInit } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { ActivatedRoute, Router } from '@angular/router';
import { TranslateService } from '@ngx-translate/core';
import {
  emailValidator,
  ErrorMessage,
  matchFieldsValidator,
} from '@weddineo-frontend/shared/ui-kit';
import { AuthFacade } from '../../+state/auth.facade';

@Component({
  selector: 'weddineo-frontend-register',
  templateUrl: './register.component.html',
  styleUrls: ['./register.component.scss'],
})
export class RegisterComponent implements OnInit {
  private readonly MIN_LENGTH: number = 8;
  formGroup: FormGroup;

  readonly errors: ErrorMessage[] = [
    {
      error: 'required',
      message: this.translateService.instant('common.validator.required'),
    },
    {
      error: 'minlength',
      message: this.translateService.instant('common.validator.minLength', {
        minLength: this.MIN_LENGTH,
      }),
    },
    {
      error: 'email',
      message: this.translateService.instant('common.validator.email'),
    },
    {
      error: 'password',
      message: this.translateService.instant('common.validator.password'),
    },
  ];

  constructor(
    private authFacade: AuthFacade,
    private translateService: TranslateService,
    private fb: FormBuilder,
    private router: Router,
    private route: ActivatedRoute
  ) {}

  ngOnInit(): void {
    this.formGroup = this.fb.group(
      {
        username: [
          null,
          [Validators.required, Validators.minLength(this.MIN_LENGTH)],
        ],
        email: [
          null,
          [
            Validators.required,
            Validators.minLength(this.MIN_LENGTH),
            emailValidator,
          ],
        ],
        password: [
          null,
          [Validators.required, Validators.minLength(this.MIN_LENGTH)],
        ],
        passwordConfirm: [
          null,
          [Validators.required, Validators.minLength(this.MIN_LENGTH)],
        ],
      },
      {
        validator: [matchFieldsValidator('password', 'passwordConfirm')],
      }
    );
  }

  register() {
    this.authFacade.register({
      username: this.formGroup.get('username').value,
      email: this.formGroup.get('email').value,
      password: this.formGroup.get('password').value,
    });
  }

  goToLogin() {
    this.router.navigate(['../'], { relativeTo: this.route });
  }
}
