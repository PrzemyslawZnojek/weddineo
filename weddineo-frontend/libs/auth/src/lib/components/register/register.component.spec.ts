import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FlexLayoutModule } from '@angular/flex-layout';
import { ReactiveFormsModule } from '@angular/forms';
import { MatCardModule } from '@angular/material/card';
import { RouterTestingModule } from '@angular/router/testing';
import { TranslatePipe, TranslateService } from '@ngx-translate/core';
import {
  ActionButtonComponent,
  TextInputComponent,
} from '@weddineo-frontend/ui-kit';
import { MockComponents, MockPipe } from 'ng-mocks';
import { AuthFacade } from '../../+state/auth.facade';
import { RegisterComponent } from './register.component';

describe('RegisterComponent', () => {
  let component: RegisterComponent;
  let fixture: ComponentFixture<RegisterComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        MatCardModule,
        FlexLayoutModule,
        ReactiveFormsModule,
        RouterTestingModule,
      ],
      declarations: [
        RegisterComponent,
        MockComponents(ActionButtonComponent, TextInputComponent),
        MockPipe(TranslatePipe),
      ],
      providers: [
        {
          provide: TranslateService,
          useClass: class {
            instant() {}
          },
        },
        {
          provide: AuthFacade,
          useClass: class {
            register() {}
          },
        },
      ],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(RegisterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should render elements', () => {
    fixture.detectChanges();
    const compiled = fixture.nativeElement;
    expect(
      compiled.querySelectorAll('weddineo-frontend-text-input').length
    ).toEqual(4);
    expect(
      compiled.querySelectorAll('weddineo-frontend-action-button').length
    ).toEqual(2);
  });
});
