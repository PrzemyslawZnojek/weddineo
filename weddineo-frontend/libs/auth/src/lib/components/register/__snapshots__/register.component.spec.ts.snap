// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterComponent should match snapshot 1`] = `
<weddineo-frontend-register
  MIN_LENGTH={[Function Number]}
  authFacade={[Function useClass]}
  errors={[Function Array]}
  fb={[Function FormBuilder]}
  formGroup={[Function FormGroup]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  translateService={[Function useClass]}
>
  <div
    class="container"
    style="align-content: center; align-items: center; box-sizing: border-box; display: flex; flex-direction: row; justify-content: center; flex: 1 1 100%; max-width: 100%;"
  >
    <mat-card
      class="mat-card mat-focus-indicator"
      style="box-sizing: border-box; max-width: 80%; flex: 1 1 100%;"
    >
      <mat-card-content
        class="mat-card-content"
        style="box-sizing: border-box; display: flex; flex-direction: column;"
      >
        <mat-card-title
          class="mat-card-title"
        >
          
        </mat-card-title>
        <weddineo-frontend-text-input />
        <weddineo-frontend-text-input />
        <weddineo-frontend-text-input />
        <weddineo-frontend-text-input />
        <div
          style="box-sizing: border-box; display: flex; flex-direction: row; align-content: center; align-items: center; justify-content: space-between;"
        >
          <weddineo-frontend-action-button
            style="align-content: center; align-items: center; box-sizing: border-box; display: flex; flex-direction: row; justify-content: center;"
          />
          <weddineo-frontend-action-button
            style="align-content: center; align-items: center; box-sizing: border-box; display: flex; flex-direction: row; justify-content: center;"
          />
        </div>
      </mat-card-content>
    </mat-card>
  </div>
</weddineo-frontend-register>
`;
